source/common/config/utility.cc

```diff
@@ -254,25 +256,49 @@ envoy::api::v2::ClusterLoadAssignment Utility::translateClusterHosts(
   return load_assignment;
 }
 
+namespace {
+absl::string_view ProtoTypeUrlToDescriptorFullName(absl::string_view type_url) {
+  size_t pos = type_url.find_last_of('/');
+  if (pos != absl::string_view::npos) {
+    type_url = type_url.substr(pos + 1);
+  }
+  return type_url;
+}
+} // namespace
+
 void Utility::translateOpaqueConfig(const ProtobufWkt::Any& typed_config,
                                     const ProtobufWkt::Struct& config,
                                     ProtobufMessage::ValidationVisitor& validation_visitor,
                                     Protobuf::Message& out_proto) {
   static const std::string& struct_type =
       ProtobufWkt::Struct::default_instance().GetDescriptor()->full_name();
+  static const std::string& typed_struct_type =
+      udpa::type::v1::TypedStruct::default_instance().GetDescriptor()->full_name();
```

Comment:
FWIW, not super happy about these being `static` non-POD, I know it was there already though. Can we make them `const char*`?

Reply:
@htuch it doesn't make any difference, a const-ref is already trivial, and the string is allocated inside descriptor pool.

Reply:
I looked at it again. I think using either reference or pointer is a gotcha, as you need to know that the descriptor pool keeps these names around. I've removed the reference std::string as it should be cleaner and not affect performance.